File: benchmark/results/java-multilang/run-003/b-baseline/output/com/metaengine/demo/billing/services/InvoiceRepository.java
package com.metaengine.demo.billing.services;

import com.metaengine.demo.billing.aggregates.Invoice;
import java.util.List;

/** InvoiceRepository domain service for the billing domain. */
public class InvoiceRepository {
    /** Persist a new Invoice from a partial input. */
    public Invoice create(Invoice input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Find an Invoice by its identifier. May return null. */
    public Invoice findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List Invoices up to the given limit. */
    public List<Invoice> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Delete an Invoice by its identifier. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: benchmark/results/python-invocation-script/run-004/b-baseline/output/metaengine_demo/billing/services/invoice_repository.py
"""InvoiceRepository service module."""
from typing import Optional

from metaengine_demo.billing.aggregates.invoice import Invoice


class InvoiceRepository:
    """InvoiceRepository service."""

    def create(self, input: Optional[Invoice]) -> Invoice:
        """Create an Invoice."""
        raise NotImplementedError("not implemented")

    def find_by_id(self, id: str) -> Optional[Invoice]:
        """Find an Invoice by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: int) -> list[Invoice]:
        """List Invoices."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete an Invoice by id."""
        raise NotImplementedError("not implemented")
